lambdas/personInfo/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/PersonInfoHandler.java

```diff
@@ -0,0 +1,188 @@
+            }
+
+            // Lambda Lifetime
+            long runTimeDuration =
+                    System.currentTimeMillis() - FUNCTION_INIT_START_TIME_MILLISECONDS;
+            Duration duration = Duration.of(runTimeDuration, ChronoUnit.MILLIS);
+            String formattedDuration =
+                    String.format(
+                            "%d:%02d:%02d",
+                            duration.toHours(), duration.toMinutesPart(), duration.toSecondsPart());
+            LOGGER.info(
+                    "Lambda {}, Lifetime duration {}, {}ms",
+                    context.getFunctionName(),
+                    formattedDuration,
+                    runTimeDuration);
+
+            LOGGER.info("Handling Person Info request");
+            Map<String, String> headers = input.getHeaders();
+            final String sessionId = headers.get("session_id");
+
+            // QuerySessionTable to get sessionItem
+            SessionItem sessionItem = sessionService.getSession(sessionId);
+
+            // throw error if session is not found
+            if (sessionItem == null || sessionItem.getSessionId() == null) {
+                throw new SessionNotFoundException("Session is not found");
+            }
+
+            // Get context from sessionItem
+            String coreContext = sessionItem.getContext();
+            LOGGER.info("Context Value = {}", coreContext);
+
+            APIGatewayProxyResponseEvent emptyResponse = new APIGatewayProxyResponseEvent();
+            emptyResponse.setStatusCode(HttpStatusCode.NO_CONTENT);
+
+            if (coreContext != null) {
+                if (coreContext.equalsIgnoreCase("check_details")) {
+                    ObjectMapper objectMapper =
+                            new ObjectMapper().registerModule(new JavaTimeModule());
+
+                    LOGGER.info("Querying PersonIdentity table with sessionId...");
+                    LOGGER.info("SessionIDUUID = {}", UUID.fromString(sessionId));
+                    var personIdentityDetailed =
+                            personIdentityService.getPersonIdentityDetailed(
+                                    UUID.fromString(sessionId));
+
+                    LOGGER.info("PersonIdentity Retrieved");
+                    eventProbe.counterMetric(Definitions.LAMBDA_PERSON_INFO_CHECK_COMPLETED_OK);
+
+                    return ApiGatewayResponseGenerator.proxyJsonResponse(
+                            HttpStatusCode.OK, personIdentityDetailed);
+                } else {
+                    LOGGER.info("Invalid Context field value, returning empty response");
+                    eventProbe.counterMetric(Definitions.LAMBDA_PERSON_INFO_CHECK_COMPLETED_OK);
+                    return emptyResponse;
+                }
+            }
+
+            LOGGER.info("Context value null, returning empty response");
+            eventProbe.counterMetric(Definitions.LAMBDA_PERSON_INFO_CHECK_COMPLETED_OK);
```

Comment:
As above should result in an LAMBDA_PERSON_INFO_CHECK_COMPLETED_ERROR but in the exception handler retun as with a oauth invalid_request